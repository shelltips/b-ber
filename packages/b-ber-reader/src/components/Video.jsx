import React from 'react'
import PropTypes from 'prop-types'
import Media from './Media'

class Video extends Media {
    static propTypes = {
        'data-autoplay': PropTypes.bool.isRequired,
        controls: PropTypes.bool.isRequired,
    }
    static contextTypes = {
        spreadIndex: PropTypes.number,
        columnGap: PropTypes.number,
        translateX: PropTypes.number,
        paddingLeft: PropTypes.number,
        paddingRight: PropTypes.number,
        transitionSpeed: PropTypes.number,
    }
    render() {
        return (
            <video // eslint-disable-line jsx-a11y/media-has-caption
                ref={node => (this.media = node)}
                onCanPlay={this.handleOnCanPlay}
                onClick={this.handleOnClick}
                {...this.props}
            >
                {/* <track kind='captions' src='sampleCaptions.vtt' srcLang='en' /> */}
                {this.props.children}
            </video>
        )
    }
}

export default Video
