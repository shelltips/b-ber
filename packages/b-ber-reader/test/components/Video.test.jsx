/* global jest,test,expect */

import React from 'react'
import renderer from 'react-test-renderer'
import Video from '../../src/components/Video'

const createOptions = () => ({
    createNodeMock: (element) => {
        if (element.type === 'video') {
            return document.createElement('video')
        }
        return null
    },
})

describe('Video', () => {
    test('renders the component', () => {

        let props
        let tree

        props = {'data-autoplay': true}
        tree = renderer.create(<Video {...props} />, createOptions()).toJSON()
        expect(tree).toMatchSnapshot()

        props = {'data-autoplay': false}
        tree = renderer.create(<Video {...props} />, createOptions()).toJSON()
        expect(tree).toMatchSnapshot()

    })
})
