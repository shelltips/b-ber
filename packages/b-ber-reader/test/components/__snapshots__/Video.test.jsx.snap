// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders the component 1`] = `
<video
  controls={true}
  data-autoplay={true}
  onCanPlay={[Function]}
  onClick={[Function]}
/>
`;

exports[`Video renders the component 2`] = `
<video
  controls={false}
  data-autoplay={false}
  onCanPlay={[Function]}
  onClick={[Function]}
/>
`;
